pkg/util/file_indexer.go

```diff
@@ -247,6 +244,39 @@ func RunIndexer(directory string, ignoreRules []string) (ret IndexerRet, err err
 	return ret, nil
 }
 
+// CalculateFileDataKeyFromPath converts an absolute path to relative (and converts to OS-specific paths) for use
+// as a map key in IndexerRet and FileIndex
+func CalculateFileDataKeyFromPath(absolutePath string, rootDirectory string) (string, error) {
```

Comment:
Unit tests for `CalculateFileDataKeyFromPath` and `GenerateNewFileDataEntry`?

Reply:
Done.